src/lib.rs

```diff
@@ -155,7 +169,7 @@ impl Build {
             // Avoid multilib-postfix for build targets that specify it
             .arg("--libdir=lib");
 
-        if cfg!(not(feature = "legacy")) {
+        if cfg!(not(feature = "legacy")) && !target.contains("wasm") {
```

Comment:
Why is this condition here added for wasm?

Reply:
> Why is this condition here added for wasm?

Ah my bad, i accidentally left that condition there.

Reply:
> No this repository only has one version of OpenSSL via a submodule, which is currently 3+

Yes this repo only contains openssl 3, but right now only OpenSSL 1.1.1d is supported with specific patches for wasm32-wasi, see https://github.com/alexcrichton/openssl-src-rs/pull/113#issuecomment-993719338

Reply:
If that's the case then I don't think this can land right now.

Reply:
> If that's the case then I don't think this can land right now.

I will try patching the latest OpenSSL 3 and see if this works with that or not.